src/skin/skinloader.cpp

```diff
@@ -213,5 +233,117 @@ SkinPointer SkinLoader::skinFromDirectory(const QDir& dir) const {
+
+    m_pShowCover = new ControlPushButton(ConfigKey("[Skin]", "show_coverart"), true);
+    m_pShowCover->setButtonMode(ControlPushButton::TOGGLE);
+
+    m_pSelectBigSpinnyCover = new ControlPushButton(
+            ConfigKey("[Skin]", "select_big_spinny_or_cover"), true);
+    m_pSelectBigSpinnyCover->setButtonMode(ControlPushButton::TOGGLE);
+
+    // This is 1 if [Skin], show_spinnies == 1 OR [Skin],show_coverart == 1
+    m_pShowSpinnyAndOrCover = new ControlPushButton(ConfigKey("[Skin]", "show_spinny_or_cover"));
+    m_pShowSpinnyAndOrCover->setButtonMode(ControlPushButton::TOGGLE);
+    m_pShowSpinnyAndOrCover->setReadOnly();
+    // This is 1 if [Skin],show_spinny_cover == 1 AND [Skin],select_big_spinny_coverart == 0
+    m_pShowSmallSpinnyCover = new ControlPushButton(
+            ConfigKey("[Skin]", "show_small_spinny_or_cover"));
+    m_pShowSmallSpinnyCover->setButtonMode(ControlPushButton::TOGGLE);
+    m_pShowSmallSpinnyCover->setReadOnly();
+    // This is 1 if [Skin],show_spinny_cover == 1 AND [Skin],select_big_spinny_coverart == 1
+    m_pShowBigSpinnyCover = new ControlPushButton(ConfigKey("[Skin]", "show_big_spinny_or_cover"));
+    m_pShowBigSpinnyCover->setButtonMode(ControlPushButton::TOGGLE);
+    m_pShowBigSpinnyCover->setReadOnly();
+
+    connect(m_pShowSpinny,
+            &ControlObject::valueChanged,
+            this,
+            &SkinLoader::updateSpinnyCoverControls);
+    connect(m_pShowCover,
+            &ControlObject::valueChanged,
+            this,
+            &SkinLoader::updateSpinnyCoverControls);
+    connect(m_pSelectBigSpinnyCover,
+            &ControlObject::valueChanged,
+            this,
+            &SkinLoader::updateSpinnyCoverControls);
+
+    m_spinnyCoverControlsCreated = true;
+    updateSpinnyCoverControls();
+}
+
+void SkinLoader::updateSpinnyCoverControls() {
+    if (!m_spinnyCoverControlsCreated) {
+        return;
+    }
+    m_pShowSpinnyAndOrCover->setAndConfirm(
+            (m_pShowSpinny->toBool() || m_pShowCover->toBool())
+                    ? 1.0
+                    : 0.0);
+    m_pShowSmallSpinnyCover->setAndConfirm(
+            (m_pShowSpinnyAndOrCover->toBool() && !m_pSelectBigSpinnyCover->toBool())
+                    ? 1.0
+                    : 0.0);
+    m_pShowBigSpinnyCover->setAndConfirm(
+            (m_pShowSpinnyAndOrCover->toBool() &&
+                    m_pSelectBigSpinnyCover->toBool())
+                    ? 1.0
+                    : 0.0);
+}
+
+void SkinLoader::setupMicDuckingControls() {
+    if (m_micDuckingControlsCreated) {
+        return;
+    }
+    // This is 1 if at least one microphone device is configured
+    m_pShowDuckingControls = new ControlPushButton(ConfigKey("[Skin]", "show_ducking_controls"));
+    m_pShowDuckingControls->setButtonMode(ControlPushButton::TOGGLE);
+    m_pShowDuckingControls->setReadOnly();
+
+    m_pNumMics = new ControlProxy("[Master]", "num_microphones", this);
```

Comment:
```suggestion
    m_pNumMics = make_parented<ControlProxy>("[Master]", "num_microphones", this);
```
To get around the plain new()